pkg/tsdb/mssql/mssql.go

```diff
@@ -1,50 +1,112 @@
 package mssql
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"net/url"
 	"reflect"
 	"regexp"
 	"strconv"
 	"strings"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
 	"github.com/grafana/grafana-plugin-sdk-go/data/sqlutil"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/grafana/grafana/pkg/util"
 
 	mssql "github.com/denisenkom/go-mssqldb"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/tsdb/sqleng"
 )
 
 var logger = log.New("tsdb.mssql")
 
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func NewExecutor(datasource *models.DataSource) (plugins.DataPlugin, error) {
-	cnnstr, err := generateConnectionString(datasource)
-	if err != nil {
-		return nil, err
+type Service struct {
+	im instancemgmt.InstanceManager
+}
+
+func ProvideService(manager backendplugin.Manager) (*Service, error) {
+	s := &Service{
+		im: datasource.NewInstanceManager(newInstanceSettings()),
 	}
-	// TODO: Don't use global
-	if setting.Env == setting.Dev {
-		logger.Debug("getEngine", "connection", cnnstr)
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: s,
+	})
+
+	if err := manager.RegisterAndStart(context.Background(), "mssql", factory); err != nil {
```

Comment:
Think @wbrowne suggested using `Register` for core plugins instead of `RegisterAndStart`.

```suggestion
	if err := manager.Register("mssql", factory); err != nil {
```